src/content/app/tools/blast/state/blast-results/blastResultsSlice.ts

```diff
@@ -62,6 +83,14 @@ export const restoreBlastSubmissions = createAsyncThunk(
   () => getAllBlastSubmissions() || {}
 );
 
+export const deleteBlastSubmission = createAsyncThunk(
+  'blast-results/delete-blast-submissions',
+  async (submissionId: string) => {
+    await deleteBlastSubmissionFromStorage(submissionId);
+    return submissionId;
```

Comment:
do we need to return submissionId? Where is it used?

Reply:
> do we need to return submissionId?

We do. Whatever is returned from a function passed to `createAsyncThunk` becomes the payload of the fulfilled action created by it.

> Where is it used?

Lines 116-117 of this file below.